tests/tests/shader/array_size_overrides.rs

```diff
@@ -11,6 +11,9 @@ const SHADER: &str = r#"
     @group(0) @binding(0)
     var<storage, read_write> output: array<u32>;
 
+    // overlaps arr's type when n is overridden to 14 and the type array<u32, 14 - 2> already exists
```

Comment:
```suggestion
    // When `n` is overridden to 14 above, it will generate the type `array<u32, 14 - 2>` which
    // already exists from this type. Ensures naga does not panic when this case happens.
    //
    // See https://github.com/gfx-rs/wgpu/issues/6762 for more info.
```

A little more verbose :)